app/controllers/cohort_imports_controller.rb

```diff
@@ -28,25 +27,43 @@ def create
 
     @cohort_import.process!
 
-    @cohort_import.record!
+    if @cohort_import.processed_only_exact_duplicates?
+      render :duplicates
+      return
```

Comment:
If you want to be fancy:

```suggestion
      render :duplicates and return
```